corehq/apps/users/models.py

```diff
@@ -538,7 +539,10 @@ def add_domain_membership(self, domain, timezone=None, **kwargs):
                 return
 
         domain_obj = Domain.get_by_name(domain, strict=True)
-        if not domain_obj:
+        _soft_assert = soft_assert(notify_admins=True)
+        if not _soft_assert(domain_obj,
+                            "Domain membership added before domain created",
+                            {'domain': domain}):
```

Comment:
Kind of unrelated. I found this when I was trying to add domain memberships to a web user from a SimpleTestCase, and it was trying to access couch just to mutate the model to create a Domain object if one didn't exist, which seemed like the wrong way to go about it. Adding the soft assert because I don't think any of our code relies on this anymore.